Build reports Maven dependencies are using the wrong scope

### Description of the bug / feature
The snapshot build reports
```
[09:39:36] :	 [Step 5/6] [INFO] --- maven-plugin-plugin:3.6.2:descriptor (default-descriptor) @ flow-maven-plugin ---
[09:39:36]W:	 [Step 5/6] [ERROR] 
[09:39:36] :	 [Step 5/6] 
[09:39:36] :	 [Step 5/6] Maven dependencies of Maven Plugins should be in provided scope.
[09:39:36] :	 [Step 5/6] Please make sure that all your dependencies declared in POM whose group ID is
[09:39:36] :	 [Step 5/6] org.apache.maven have set '<scope>provided</scope>' as well.
[09:39:36] :	 [Step 5/6] In the future this error will break the build.
[09:39:36] :	 [Step 5/6] 
[09:39:36] :	 [Step 5/6] The following dependencies are in wrong scope:
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-plugin-api:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-model:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-artifact:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-core:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-settings:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-settings-builder:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-builder-support:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-repository-metadata:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-model-builder:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6]  * org.apache.maven:maven-resolver-provider:jar:3.8.4:compile
[09:39:36] :	 [Step 5/6] 
[09:39:36] :	 [Step 5/6] Please fix your build!
[09:39:36] :	 [Step 5/6] 
```

--- a/flow-plugins/flow-maven-plugin/pom.xml
+++ b/flow-plugins/flow-maven-plugin/pom.xml
@@ -26,11 +26,13 @@
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-plugin-api</artifactId>
             <version>${maven.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-core</artifactId>
             <version>${maven.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.google.gwt</groupId>

